vmdb/db/fixtures/ae_datastore/ManageIQ/Service/Provisioning/StateMachines/Methods.class/__methods__/catalogbundleinitialization.rb

```diff
@@ -68,14 +68,23 @@ def tag_service(dialogs_tags_hash)
   log_and_update_message(:info, "Processing tag_service...Complete", true)
 end
 
-def parsed_dialog_information
-  # get parsed dialog options and tags from the task
-  dialog_options_hash = YAML.load(@task.get_option(:parsed_dialog_options))
-  dialog_tags_hash = YAML.load(@task.get_option(:parsed_dialog_tags))
+def dialog_parser_error
+  log_and_update_message(:error, "Error loading dialog options")
+  exit MIQ_ABORT
+end
 
+def yaml_data(option)
+  YAML.load(@task.get_option(option))
+end
+
+def parsed_dialog_information
+  dialog_options_hash = @task.get_option(:parsed_dialog_options).nil? ? nil : yaml_load(:parsed_dialog_options)
+  dialog_tags_hash = @task.get_option(:parsed_dialog_tags).nil? ? nil : yaml_load(:parsed_dialog_tags)
```

Comment:
You meant to call the `yaml_data` method on these two lines.  Also, since the yaml_data method is designed to get the data from the task's options hash I would move the nil check logic into that method as well.